UsefulProteomicsDatabases/ProteinDbLoader.cs

```diff
@@ -515,6 +515,64 @@ public static List<Protein> LoadProteinFasta(string proteinDbLocation, bool onTh
             return result;
         }
 
+        /// <summary>
+        /// Merges proteins of the same accession and sequence. Considers contaminants separately.
+        /// </summary>
+        /// <param name="merge_these"></param>
+        /// <returns></returns>
+        public static IEnumerable<Protein> merge_proteins(IEnumerable<Protein> merge_these)
+        {
+            Dictionary<Tuple<string, string, bool, bool>, List<Protein>> proteins_by_accession_seq_cont_isdecoy = new Dictionary<Tuple<string, string, bool, bool>, List<Protein>>();
+            foreach (Protein p in merge_these)
+            {
+                Tuple<string, string, bool, bool> key = new Tuple<string, string, bool, bool>(p.Accession, p.BaseSequence, p.IsContaminant, p.IsDecoy);
+                if (!proteins_by_accession_seq_cont_isdecoy.TryGetValue(key, out List<Protein> bundled))
+                    proteins_by_accession_seq_cont_isdecoy.Add(key, new List<Protein> { p });
+                else
+                    bundled.Add(p);
+            }
```

Comment:
Good question. I think this might be intended in some cases. In Rachel K's work, she is hard-coding in sequence variations for different HIV strains, but leaving the accessions the same. Leaving the accessions the same for different sequences is important for quantification, where they should be considered the same "protein."